### PR TITLE
Merge Sort in JavaScript

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Merge Sort in JavaScript

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1427

### CODE REVIEW

File: JavaScript/sort/MergeSort.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  // Base case or terminating case
+  if (array.length < 2) {
+    return array;
+  }
+
+  const left = array.splice(0, half);
+  return merge(mergeSort(left), mergeSort(array));
+}
+
+array = [4, 8, 7, 2, 11, 1, 3];
```
Comment (HarshCasper) [inline JavaScript/sort/MergeSort.js]: Make the input dynamic.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny change. Ready to merge then 🍾

File: JavaScript/sort/MergeSort.js (hunk 1 above)
Comment (Siddhant-K-code) [reply JavaScript/sort/MergeSort.js]: Fixed! Please Review It Again.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Siddhant-K-code Add space and time complexity.

**Comment by Siddhant-K-code** [comment]
> @Siddhant-K-code Add space and time complexity.

Done, Pls review @iamrajiv @HarshCasper

**Review by iamrajiv — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**